lib/tracking_options.dart

```diff
@@ -0,0 +1,92 @@
+import 'constants.dart';
+
+/// Options to control the values tracked in SDK
+///
+/// Different platforms have different default events.
+/// Refer to platform specific docs for more details.
+///
+/// * [iOS](https://www.docs.developers.amplitude.com/data/sdks/ios-swift/#disable-tracking)
+/// * [Android](https://www.docs.developers.amplitude.com/data/sdks/android-kotlin/#disable-tracking)
+/// * [Web](https://www.docs.developers.amplitude.com/data/sdks/browser-2/#optional-tracking)
+class TrackingOptions {
+  final bool disableIpAddress;
+  final bool disableLanguage;
+  final bool disablePlatform;
+  // Mobile (iOS and Android) specific
+  final bool disableRegion;
+  final bool disableDma;
+  final bool disableCountry;
+  final bool disableCity;
+  final bool disableCarrier;
+  final bool disableDeviceModel;
+  final bool disableDeviceManufacturer;
+  final bool disableOsVersion;
+  final bool disableOsName;
+  final bool disableVersionName;
+  // Android specific
+  final bool disableADID;
+  final bool disableAppSetId;
+  final bool disableDeviceBrand;
+  final bool disableLatLag;
+  final bool disableApiLevel;
+  // iOS specific
+  final bool disableIDFV;
```

Comment:
I think I prefer the Browser style config here without `disabled` prefix. I think the `disableXXX()` is useful in the function names to clarify the action, but as part of the config I think it is more concise/understandable to just use the option name directly e.g. `ipAddress`

e.g.

```
amplitude.init(API_KEY, {
  trackingOptions: {
    ipAddress: false,
    language: false,
    platform: false,
  },
});
```

Reply:
Makes sense. See https://github.com/amplitude/Amplitude-Flutter/pull/162/commits/7624e9953d0cafd00527c45608fe999e7aa431bb